src/functionalTest/kotlin/kotlinx/kover/test/functional/cases/VerificationTests.kt

```diff
@@ -57,55 +57,93 @@ internal class VerificationTests : BaseGradleScriptTest() {
-                                "    covered lines count is 4, but expected maximum is 3"
-                    )
-                }
-            }
-        }
-    }
-
-    @Test
-    fun testNotVerifiedJaCoCo() {
-        val build = diverseBuild(languages = ALL_LANGUAGES, engines = listOf(CoverageEngineVendor.JACOCO))
-        build.addKoverRootProject {
-            sourcesFrom("simple")
-
-            kover {
-                verify {
                     rule {
-                        name = "test rule"
+                        name = "fully uncovered instructions by classes"
+                        target = VerificationTarget.CLASS
                         bound {
-                            minValue = 58
-                            maxValue = 60
+                            counter = CounterType.INSTRUCTION
+                            valueType = VerificationValueType.MISSED_PERCENTAGE
+                            minValue = 100
                         }
+                    }
+                    rule {
+                        name = "fully covered instructions by packages"
+                        target = VerificationTarget.PACKAGE
                         bound {
+                            counter = CounterType.INSTRUCTION
+                            valueType = VerificationValueType.COVERED_PERCENTAGE
+                            minValue = 100
+                        }
+                    }
+                    rule {
+                        name = "branches by classes"
+                        target = VerificationTarget.CLASS
+                        bound {
+                            counter = CounterType.BRANCH
                             valueType = VerificationValueType.COVERED_COUNT
-                            minValue = 2
-                            maxValue = 3
+                            minValue = 1000
+                        }
+                    }
+                    rule {
+                        name = "missed packages"
+                        target = VerificationTarget.PACKAGE
+                        bound {
+                            valueType = VerificationValueType.MISSED_COUNT
+                            maxValue = 1
                         }
                     }
                 }
             }
         }
 
-        build.prepare().runWithError("koverVerify") {
-            output {
-                assertTrue {
-                    this.contains(
-                        "[ant:jacocoReport] Rule violated for bundle :: lines covered ratio is 0.50, but expected minimum is 0.58\n" +
-                                "[ant:jacocoReport] Rule violated for bundle :: lines covered count is 4, but expected maximum is 3"
-                    )
-                }
+        build.prepare().runWithError("koverHtmlReport", "koverVerify") {
+            verification {
+                assertIntelliJResult("""Rule 'counts rule' violated:
```

Comment:
Works for me, but in case when there is more than one such test, I would suggest extracting test data (these long strings) into `test-resources` text files available by `@TestName` similarly to this:

Reply:
I think yes, with an increase in the number of verification tests, these lines should be put into files.